### PR TITLE
refactor: remove unused error paths, use expect when possible

### PR DESCRIPTION
**Pull request by joschisan**

### CODE REVIEW

File: modules/fedimint-mint-client/src/output.rs [hunk 1]
```diff
@@ -220,68 +212,66 @@ impl MintOutputStatesCreated {
...
+                .into_iter()
+                .map(|(peer, share)| (peer.to_usize() as u64 + 1, share))
+                .collect(),
+        );
+
+        let amount_key = tbs_pks
+            .tier(&created.amount)
+            .expect("We obtained this amount from tbs_pks when we created the output");
+
+        // this implies that the mint client config's public keys are inconsistent
```
Comment (elsirion) [inline modules/fedimint-mint-client/src/output.rs:238]: So maybe in the future we can just check config consistency when joining the federation and then expect here?

**Review by elsirion — APPROVED** [review]
Great refactor!

File: modules/fedimint-mint-client/src/output.rs (hunk 1 above)
Comment (joschisan) [reply modules/fedimint-mint-client/src/output.rs:238]: Yes we could

File: crypto/tbs/src/lib.rs [hunk 2]
```diff
@@ -185,6 +185,14 @@ fn lagrange_multipliers(scalars: Vec<Scalar>) -> Vec<Scalar> {
         .collect()
 }
 
+pub fn verify_blinded_signature(
```
Comment (m1sterc001guy) [inline crypto/tbs/src/lib.rs:188]: Some docs might be useful here for everything that isnt familiar with the pairing crypto

File: modules/fedimint-mint-client/src/output.rs [hunk 3]
```diff
@@ -220,68 +212,66 @@ impl MintOutputStatesCreated {
...
-                MintOutputStateMachine {
-                    common: old_state.common,
-                    state: MintOutputStates::Succeeded(MintOutputStatesSucceeded { amount }),
-                }
-            }
-            Err(error) => MintOutputStateMachine {
+        let agg_blind_signature = aggregate_signature_shares(
+            &blinded_signature_shares
+                .into_iter()
+                .map(|(peer, share)| (peer.to_usize() as u64 + 1, share))
```
Comment (m1sterc001guy) [inline modules/fedimint-mint-client/src/output.rs:230]: +1 here is just to prevent a peer with id 0?

Comment (joschisan) [reply modules/fedimint-mint-client/src/output.rs:230]: We need the +1 here so the cryptography checks out.

**Review by m1sterc001guy — APPROVED** [review]
I'm not the most well-versed in the mint module but this looks like a good refactor to me

File: modules/fedimint-mint-client/src/output.rs (hunk 3 above)
Comment (elsirion) [reply modules/fedimint-mint-client/src/output.rs:230]: I knew what it was for but also raised an eye brow that there is no fn to convert the peer ID. But that's how I wrote it an eternity ago and Joscha just kept it that way.

**Pull request merged**